src/Equinor.ProCoSys.IPO.WebApi/Synchronization/BusReceiverService.cs

```diff
@@ -157,36 +148,38 @@ private void ProcessCommPkgEvent(string messageJson)
-                        {PcsServiceBusTelemetryConstants.ProjectName, commPkgEvent.ProjectName.Replace('$', '_')},
-                        {PcsServiceBusTelemetryConstants.ProjectNameOld, commPkgEvent.ProjectNameOld.Replace('$', '_')}
-                    });
-                _invitationRepository.MoveCommPkg(
-                    commPkgEvent.ProjectNameOld,
-                    commPkgEvent.ProjectName,
-                    commPkgEvent.CommPkgNo,
-                    commPkgEvent.Description);
-            }
-            else
+
+            // Get or create project
+            var project = _invitationRepository.GetProject(commPkgEvent.ProjectName);
+            if (project == null)
             {
-                _telemetryClient.TrackEvent(IpoBusReceiverTelemetryEvent,
-                    new Dictionary<string, string>
-                    {
-                        {PcsServiceBusTelemetryConstants.Event, IpoTopic.TopicName},
-                        {PcsServiceBusTelemetryConstants.CommPkgNo, commPkgEvent.CommPkgNo},
-                        {PcsServiceBusTelemetryConstants.Plant, commPkgEvent.Plant[4..]},
-                        {PcsServiceBusTelemetryConstants.ProjectName, commPkgEvent.ProjectName.Replace('$', '_')}
-                    });
-                _invitationRepository.UpdateCommPkgOnInvitations(commPkgEvent.ProjectName, commPkgEvent.CommPkgNo,
-                    commPkgEvent.Description);
+                var proCoSysProject = await _projectApiService.TryGetProjectAsync(commPkgEvent.Plant, commPkgEvent.ProjectName);
+                if (proCoSysProject is null)
+                {
+                    throw new IpoValidationException(
+                        $"Could not find ProCoSys project called {commPkgEvent.ProjectName} in plant {commPkgEvent.Plant}");
+                }
+
+                project = new Project(commPkgEvent.Plant, commPkgEvent.ProjectName, proCoSysProject.Description, proCoSysProject.ProCoSysGuid)
+                {
+                    IsClosed = proCoSysProject.IsClosed
+                };
+
+                _projectRepository.Add(project);
+                CancellationToken cancellationToken = default;
+                await _unitOfWork.SaveChangesAsync(cancellationToken);
             }
+
+            _telemetryClient.TrackEvent(IpoBusReceiverTelemetryEvent,
+                new Dictionary<string, string>
+                {
+                    {PcsServiceBusTelemetryConstants.Event, IpoTopic.TopicName},
+                    {PcsServiceBusTelemetryConstants.CommPkgNo, commPkgEvent.CommPkgNo},
+                    {PcsServiceBusTelemetryConstants.Plant, commPkgEvent.Plant[4..]},
+                    {PcsServiceBusTelemetryConstants.ProjectName, commPkgEvent.ProjectName.Replace('$', '_')}
+                });
+
+            _invitationRepository.UpdateCommPkgOnInvitations(commPkgEvent.ProCoSysGuid, commPkgEvent.CommPkgNo,commPkgEvent.Description,project);

```

Comment:
I think we need to make some changes in invitationRepository here as well. We are getting the commPkgs to update based on the commPkgNo and project that we are sending in as parameters. Here we are sending the new project that we are moving TO, and hence, we will not retreive the commpkgs to be updates in the _invitationRepository.UpdateCommPkgOnInvitations-method I think.

Reply:
If it is possible to update description and project in the same event

Reply:
In UpdateCommPkgOnInvitations I think we should use Guid to get the commpkg. Same when updating mcpkg